test/goose/brokers/redis/batch_test.clj

```diff
@@ -48,4 +59,106 @@
       (redis-batch/enqueue tu/redis-conn batch)
 
       (is (= (redis-cmds/parse-map tu/redis-conn batch-state-key) batch-state))
-      (is (= (redis-cmds/set-members tu/redis-conn enqueued-job-set) job-ids)))))
\ No newline at end of file
+      (is (= (redis-cmds/set-members tu/redis-conn enqueued-job-set) job-ids)))))
+
+;;;; ======= TEST: Batch state update middleware ===========
+
+(deftest enqueued-to-successful
+  (testing "Job state is transitioned from enqueued -> successful on successful job execution"
+    (redis-cmds/add-to-set (:redis-conn opts) enqueued-job-set job-id)
+    ((redis-batch/update-state (fn [_opts _job] "Function executed"))
+     opts job)
+    (is (= 0 (redis-cmds/set-size (:redis-conn opts)
+                                  enqueued-job-set)))
+    (is (= 1 (redis-cmds/set-size (:redis-conn opts)
+                                  successful-job-set)))
+    (is (= 0 (redis-cmds/set-size (:redis-conn opts)
+                                  retry-job-set)))
+    (is (= 0 (redis-cmds/set-size (:redis-conn opts)
+                                  dead-job-set)))))
+
+(deftest enqueued-to-retrying
+  (testing "Job state is transitioned from enqueued -> retrying on job failure"
```

Comment:
Can we remove max-retries from the job map & add a max-retries here like in enqueued -> dead job? 

Not a blocker, just that I was confused how this test is passing on first look & had to scroll up.

Reply:
```suggestion
  (testing "Job state is transitioned from enqueued -> retrying on job failure"
    (let [job (assoc job :retry-opts {:max-retries 1})]
```

Reply:
The reason to move the max-retries to the `def` level declaration of `job` was to remove redundant declaration of the same in other tests like `retrying-to-successful` and `retrying-to-dead`.

Reply:
Keeping max-retries at `def`-level will reduce redundancies. However, I feel explicitly defining them in test cases would be clearer to developer. 
I don't have strong opinions either way. Choose 1, and feel free to resolve the conversation.

Reply:
Akshat, I will go ahead and keep the max-retries in the `def` level. Resolving the comment.